src/modm/driver/display/ssd1306.hpp

```diff
@@ -94,40 +41,28 @@ struct ssd1306
 		Frames256 = 0b011
 	};
 
-	enum class
-	ScrollDirection : uint8_t
+	enum class ScrollDirection : uint8_t
 	{
-		Right = SetHorizontalScrollRight,
-		Left = SetHorizontalScrollLeft
+		Right = HorizontalScrollRight,
+		Left = HorizontalScrollLeft,
+		// RightBottom = VerticalAndHorizontalScrollRight,
+		// LeftBottom = VerticalAndHorizontalScrollLeft,
 	};
 
-	enum class
-	DisplayMode : uint8_t
+	enum class DisplayMode : uint8_t
 	{
-		Normal = SetNormalDisplay,
-		Inverted = SetInvertedDisplay
+		Normal = NormalDisplay,
+		Inverted = InvertedDisplay
 	};
 
-protected:
-	/// @cond
-	static constexpr uint8_t
-	i(ScrollDirection direction) { return uint8_t(direction); }
-	static constexpr uint8_t
-	i(ScrollStep step) { return uint8_t(step); }
-
 public:
-	template < uint8_t Height >
-	class DataTransmissionAdapter : public modm::I2cWriteTransaction
+	class Ssd1306_I2cWriteTransaction : public modm::I2cWriteTransaction
 	{
 	public:
-		DataTransmissionAdapter(uint8_t address);
-
-		void modm_always_inline
-		setCommandBuffer(uint8_t *buffer)
-		{ commands = buffer; }
+		Ssd1306_I2cWriteTransaction(uint8_t address);
 
 		bool
-		configureDisplayWrite(uint8_t (*buffer)[Height / 8], std::size_t size);
+		configureDisplayWrite(const uint8_t *buffer, const std::size_t size);
```

Comment:
```suggestion
		configureDisplayWrite(const uint8_t *buffer, std::size_t size);
```
Pass-by-value is always a copy, hence const is redundant.